qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -615,13 +615,16 @@ bool FitScriptGeneratorModel::checkFunctionIsSameForAllDomains() const {
   return std::all_of(m_fitDomains.cbegin(), m_fitDomains.cend(), hasSameFunction);
 }
 
-std::tuple<bool, std::string> FitScriptGeneratorModel::isValid() const {
+std::tuple<bool, std::string> FitScriptGeneratorModel::isValid(std::string const &outputBaseName) const {
```

Comment:
Just a quick question here on passing `outputBaseName` as an argument. The rest of the properties that are checked are checked through accessor functions. Could you help me understand why the base name is an argument?
My thinking at first was that `isValid` is checking the validity of the object for running the generation and if `outputBaseName` could be different when it is run is that still valid? I may have misunderstood the context here though.

Reply:
When changing the `outputBaseName` in the view, there is not a corresponding member variable in the model to update. Therefore I was passing in the `outputBaseName` to `isValid` at the point when it was needed for validation.

I think I did it this way because it was quicker to implement, and the `outputBaseName` is not needed until the point when `Generate Script` is clicked. Do you think it would be better to update the `outputBaseName` within a model field variable whenever it is changed in the view?

The `outputBaseName` can be anything apart from an empty string, I'm not sure if that answers your question?

Reply:
This answers my question. Thanks.

My thinking was that if the field is not in the model then it's possible that `isValid` can be called with a different string than would be used at generation time so `Model::isValid` isn't quite giving a full enough answer. I saw this as a bit similar to the algorithms where you have:
* create object
* set properties
* validate them as a set such that the next call to generate will work as expected

What do you think?

Reply:
I see, yes I think it would be better to store it in the model in that case. I will make some changes to this - thanks